main.tf

```
@@ -75,11 +75,69 @@ resource "aws_s3_bucket" "tf_backend_bucket" {
     ManagedByTerraform = "true"
     TerraformModule = "terraform-aws-backend"
   }
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        kms_master_key_id = "${var.kms_key_id}"
+        sse_algorithm     = "${var.sse_algorithm}"
+      }
+    }
+  }
   lifecycle {
     prevent_destroy = true
   }
 }
 
+data "aws_iam_policy_document" "tf_backend_bucket_policy" {
+  statement {
+    sid = "RequireEncryptedTransport"
+    effect = "Deny"
+    actions = [
+      "s3:*",
+    ]
+    resources = [
+      "${aws_s3_bucket.tf_backend_bucket_encrypted.arn}/*"
+    ]
+    condition {
+      test = "Bool"
+      variable = "aws:SecureTransport"
+      values = [
+        false,
+      ]
+    }
+    principals = [
+      "*",
+    ]
+  }
+
+  statement {
+    sid = "RequireEncryptedStorage"
+    effect = "Deny"
+    actions = [
+      "s3:PutObject",
+    ]
+    resources = [
+      "${aws_s3_bucket.tf_backend_bucket_encrypted.arn}/*"
+    ]
+    condition {
+      test = "StringNotEquals"
+      variable = "s3:x-amz-server-side-encryption"
+      values = [
+        "AES256"
+      ]
+    }
```

Outstanding questions... will this bucket policy block requests if the SSE Algorithm is set to `AES256`? Need to check that and make sure that either algorithm will work with this policy, OR.... is this crazy?

```suggestion
    condition {
      test = "StringEquals"
      variable = "s3:x-amz-server-side-encryption"
      values = [
        "${var.sse_algorithm}"
      ]
    }
```

---

I think you can deny unencrypted storage by doing this:
```
      "Condition": {
        "Null": {
          "s3:x-amz-server-side-encryption": "true"
        }
      }
```

source: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html

---

OK, here is where i landed:
1) it is stronger to "reject invalid cases". so the existing "deny if not equal to" style rule will persist
2) i did miss getting the var in here, i have corrected this.